From 5f69ea6f29371f0d03c23ff0ec81b63b7e5dca23 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@theqtcompany.com>
Date: Thu, 9 Jun 2016 13:18:36 +0200
Subject: [PATCH] Use -fno-delete-null-pointer-checks with gcc 6

V8 is not safe for removing null pointer checks. Until it is, we need
to disable optimizations in GCC that assume it is not doing something
wrong.

Task-number: QTBUG-53956
Change-Id: Ieae1a44a36ddd481656497414275b134bbbc9da3
---
 chromium/build/common.gypi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/chromium/build/common.gypi b/chromium/build/common.gypi
index 272a42f..209ffe0 100644
--- a/chromium/build/common.gypi
+++ src/3rdparty/chromium/build/common.gypi
@@ -4729,6 +4729,26 @@
               }],
             ],
           }],
+          ['gcc_version>=60 and clang==0', {
+            'target_conditions': [
+              ['_toolset=="target"', {
+                'cflags_cc': [
+                  # V8 needs this, see https://bugs.chromium.org/p/v8/issues/detail?id=3782
+                  '-fno-delete-null-pointer-checks',
+                ],
+              }],
+            ],
+          }],
+          ['host_gcc_version>=60 and clang==0 and host_clang==0', {
+            'target_conditions': [
+              ['_toolset=="host"', {
+                'cflags_cc': [
+                  # V8 needs this, see https://bugs.chromium.org/p/v8/issues/detail?id=3782
+                  '-fno-delete-null-pointer-checks',
+                ],
+              }],
+            ],
+          }],
         ],
       },
     }],
-- 
2.8.3

